packages/backend-defaults/src/entrypoints/urlReader/lib/GoogleGcsUrlReader.ts

```diff
@@ -121,8 +133,36 @@ export class GoogleGcsUrlReader implements UrlReaderService {
     return ReadUrlResponseFactory.fromReadable(stream);
   }
 
-  async readTree(): Promise<UrlReaderServiceReadTreeResponse> {
-    throw new Error('GcsUrlReader does not implement readTree');
+  async readTree(
+    url: string,
+    _options?: UrlReaderServiceReadTreeOptions,
+  ): Promise<UrlReaderServiceReadTreeResponse> {
+    const { bucket, key } = parseURL(url);
+
+    if (key.match(/[*?]/)) {
+      throw new Error(
+        'GcsUrlReader readTree does not support glob patterns, use search instead',
+      );
+    }
+
+    const [files] = await this.storage.bucket(bucket).getFiles({
+      autoPaginate: true,
+      prefix: key,
+    });
+
+    const responses = [];
+    for (const file of files) {
+      const buffer = await getRawBody(file.createReadStream());
```

Comment:
Is this file stream opened for each file in the bucket? Is it possible to defer it until the `content()` is called on the response item?

Not a blocker but nice optimization if possible